engine/src/object.cpp

```diff
@@ -5784,7 +5787,32 @@ void MCDeletedObjectsOnObjectResumeDeletion(MCObject *p_object, void *p_deletion
 	if (p_deletion_cookie == nil)
 		return;
 	
-	p_object -> setdeletedobjectpool((MCDeletedObjectPool *)p_deletion_cookie);
+	MCDeletedObjectPool *t_pool = (MCDeletedObjectPool*)p_deletion_cookie;
+	MCDeletedObjectPool *t_current_pool = p_object->getdeletedobjectpool();
+	
+	if (t_current_pool != nil)
+	{
+		// Restore the link to the saved deleted object pool, removing any reference
+		//    to the pool currently set on the object.
+		t_current_pool->references -= 1;
+
+		p_object -> setdeletedobjectpool(t_pool);
+	}
+	else
+	{
+		// If the current pool is nil then the object must have been deleted, so
+		//    remove the reference to the saved deleted object pool and clean up.
+		t_pool->references -= 1;
```

Comment:
It looks like we do this more than once - perhaps refactor into `MCDeletedObjectPoolRelease`?